File: leetcode/0150-evaluate-reverse-polish-notation/solution.java
class Solution {
    public int evalRPN(String[] tokens) {

        Stack<Integer> stack = new Stack<>();
        int a, b;
        for(String s : tokens){
            if(s.equals("+")){
                stack.add(stack.pop() + stack.pop());
            }
            else if(s.equals("-")){
                b = stack.pop();
                a = stack.pop();
                stack.add(a - b);
            }
            else if(s.equals("*")){
                stack.add(stack.pop() * stack.pop());
            }
            else if(s.equals("/")){
                b = stack.pop();
                a = stack.pop();
                stack.add(a / b);
            }
            else{
                stack.add(Integer.parseInt(s));
            }
            
        }
        return stack.pop();
    }
}

File: leetcode/0150-evaluate-reverse-polish-notation/solution.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for token in tokens:
            if token not in "+-*/":
                stack.append(int(token))
            else:
                n1 = stack.pop()
                n2 = stack.pop()
                if token == '+':
                    stack.append(n1 + n2)
                elif token == '-':
                    stack.append(n2 - n1)
                elif token == '*':
                    stack.append(n1 * n2)
                elif token == '/':
                    stack.append(int(n2 / n1))
        
        return stack[-1]

